src/core/AutoTypeAssociations.cpp

```diff
@@ -106,7 +106,7 @@ int AutoTypeAssociations::size() const
 int AutoTypeAssociations::associationsSize() const
 {
     int size = 0;
-    foreach (const Association &association, m_associations) {
+    for (const Association &association : m_associations) {
```

Comment:
& belongs to the type. You also need to cast m_associations to const using asConst() from core/global.h. Otherwise the container will detach and you will create unnecessary copies.

Reply:
I'm not quite sure the `asConst` is required at this point since the method itself is annotated with const making (as far as I know) m_associations const as well. I'll look into this and adjust the code accordingly.

Reply:
True, discard my comment.